### PR TITLE
get_leaderboard

### PR DESCRIPTION
**Pull request by moezali1**
I have recently implemented a new function called `get_leaderboard`. This function retrieves all the trained model object and their metrics from global variables and returns as `pandas.DataFrame`. This is how the desired output looks like:

image

I created this function in my Notebook and used the `get_config` function to retrieve everything I needed. Here is my code:

```
import pandas as pd
from pycaret.internal.tabular import _get_model_name

def get_leaderboard():
    
    model_container = get_config('master_model_container')
    result_container = get_config('create_model_container')
    
    result_container_mean = []
    
    mc = 0

    for i in np.arange(0,len(result_container)):
        c = result_container[i]
        r = c[-2:-1]
        model_name = _get_model_name(model_container[mc]) 
        r['Model Name'] = model_name
        r['Model'] = model_container[mc]
        r.set_index('Model Name', drop=True, inplace=True)
        r = r[['Model', 'Accuracy', 'AUC', 'Recall', 'Prec.', 'F1', 'Kappa', 'MCC']]
        result_container_mean.append(r)
        mc += 1
    
    results = pd.concat(result_container_mean)
    
    results.reset_index(inplace=True)
    
    return results
```

The main problem is when I execute this code directly the function that I pushed in `tabular.py` I get this error:

image

You can reproduce this error from the `get_leaderboard` branch or this draft PR:

```
from pycaret.datasets import get_data
data = get_data('juice')

from pycaret.classification import *
s = setup(data, target = 'Purchase', session_id = 123)

best = compare_models()

get_leaderboard()
```

Also please feel free to improve the code if you would like. Let me know if you have any questions. 

Thanks

### CODE REVIEW

File: pycaret/internal/tabular.py [hunk 1]
```diff
@@ -9871,6 +9871,34 @@ def load_config(file_name: str):
...
+    model_container = get_config('master_model_container')
+    result_container = get_config('create_model_container')
+    
+    result_container_mean = []
+    
+    mc = 0
+
+    for i in np.arange(0,len(result_container)):
+        c = result_container[i]
+        r = c[-2:-1]
```
Comment (Yard1) [inline pycaret/internal/tabular.py:9887]: Could we have more informative variable name here?

File: pycaret/internal/tabular.py [hunk 2]
```diff
@@ -9871,6 +9871,34 @@ def load_config(file_name: str):
...
+        model_name = _get_model_name(model_container[mc]) 
+        r['Model Name'] = model_name
+        r['Model'] = model_container[mc]
+        r.set_index('Model Name', drop=True, inplace=True)
+        r = r[['Model', 'Accuracy', 'AUC', 'Recall', 'Prec.', 'F1', 'Kappa', 'MCC']]
```
Comment (Yard1) [inline pycaret/internal/tabular.py:9892]: We should get metrics from the result container (its columns) and not hardcode them like this

File: pycaret/internal/tabular.py [hunk 3]
```diff
@@ -9871,6 +9871,34 @@ def load_config(file_name: str):
...
+    generates leaderboard for all models run in current run.
+    """
+    model_container = get_config('master_model_container')
+    result_container = get_config('create_model_container')
+    
+    result_container_mean = []
+    
+    mc = 0
+
+    for i in np.arange(0,len(result_container)):
```
Comment (Yard1) [inline pycaret/internal/tabular.py:9885]:
```suggestion
    for i in range(len(result_container)):
```

File: pycaret/regression.py [hunk 4]
```diff
@@ -2395,3 +2395,23 @@ def load_config(file_name: str):
...
+
+
+    Example
+    -------
+    >>> from pycaret.classification import get_leaderboard
+    >>> leaderboard = get_leaderboard()
+
+
+    Returns:
+        Pandas.DataFrame
```
Comment (Yard1) [inline pycaret/regression.py:2414]:
```suggestion
        pandas.DataFrame
```

File: pycaret/classification.py [hunk 5]
```diff
@@ -2622,3 +2622,22 @@ def load_config(file_name: str):
...
+
+
+    Example
+    -------
+    >>> from pycaret.classification import get_leaderboard
+    >>> leaderboard = get_leaderboard()
+
+
+    Returns:
+        Pandas.DataFrame
```
Comment (Yard1) [inline pycaret/classification.py:2640]:
```suggestion
        pandas.DataFrame
```

**Review by Yard1 — CHANGES_REQUESTED** [review]
Great, thanks. Left a few comments

File: pycaret/internal/tabular.py (hunk 2 above)
Comment (moezali1) [reply pycaret/internal/tabular.py:9892]: Hey, This is not hardcoding. I am just re-arranging the display of columns in line 9892

**Pull request merged**